### PR TITLE
Add keycloak provider

### PR DESCRIPTION
**Pull request by voenniy**
Keycloak.com is an open source Identity and Access Management solution aimed at modern applications and services.
https://github.com/keycloak/keycloak

### CODE REVIEW

**Comment by voenniy** [comment]
> Please make sure you're also PRing to the Documentation repository

https://github.com/SocialiteProviders/Documentation/pull/77

**Comment by Jimbolino** [comment]
@m1guelpf any update on when this can be merged?

**Comment by voenniy** [comment]
Hi! Can you accept PR?

File: src/Keycloak/Provider.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id' => $user['sub'],
+            'nickname' => $user['preferred_username'],
+            'name' => $user['given_name'],
```
Comment (Jimbolino) [inline src/Keycloak/Provider.php:67]: preferred_username / given_name / email are not always available, so should be fetched with `Arr::get()`

Comment (atymic) [reply src/Keycloak/Provider.php:67]: @voenniy please use this :)

File: src/Keycloak/Provider.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        return rtrim(rtrim($this->getConfig('base_url'), '/') . '/realms/' . $this->getConfig('realms', 'master'), '/');
+    }
+
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase($this->getBaseUrl() . "/protocol/openid-connect/auth", $state);
```
Comment (atymic) [inline src/Keycloak/Provider.php:35]: Single quotes please

File: src/Keycloak/Provider.php (hunk 1 above)
Comment (Jimbolino) [reply src/Keycloak/Provider.php:67]: I've opened a pull request on his repo back in February, but got no response, shall i open a new one?

**Comment by atymic** [comment]
Thanks :) Please apply the styleci changes and I will merge.
Also please add it to the monorepo split script (in the repo root)

**Comment by atymic** [comment]
Tagged :)
https://github.com/SocialiteProviders/Keycloak/releases/tag/v3.0.0

**Pull request merged**